navigator.go

```diff
@@ -99,8 +100,7 @@ func (n navigator) Follow(rel string) navigator {
 // Followf adds a relation to the follow queue of the navigator, with a
 // set of parameters to expand on execution.
 func (n navigator) Followf(rel string, params P) navigator {
-	relations := make([]relation, 0, len(n.path)+1)
-	copy(n.path, relations)
```

Comment:
Actually, I was wondering about this because I see this append(x, y...) usage even in the Go standard library.  I suspect they do some compiler optimizations for this case.  So, I did [some benchmarks](https://gist.github.com/xogeny/b819af6a0cf8ba1caaef) that show `copy` and `append` are pretty comparable.  But then I did [some further benchmarks](https://gist.github.com/xogeny/c8dfd0c4742aee502926) and they show similar results.

I'd be curious to see if you find any flaws in those cases.